code/src/item_override.c

```diff
@@ -523,6 +538,55 @@ void ItemOverride_GetSkulltulaToken(Actor* tokenActor) {
     }
 }
 
+u8 ItemOverride_GetItemDrop(EnItem00* this) {
+    // Override items that always behave as drops and never trigger a Get Item event.
+    switch (this->actor.params) {
+        case ITEM00_RUPEE_GREEN:
+        case ITEM00_RUPEE_BLUE:
+        case ITEM00_RUPEE_RED:
+        case ITEM00_RUPEE_PURPLE:
+        case ITEM00_RUPEE_ORANGE:
+        case ITEM00_RECOVERY_HEART:
+        case ITEM00_FLEXIBLE:
+        case ITEM00_BOMBS_A:
+        case ITEM00_BOMBS_B:
+        case ITEM00_ARROWS_SINGLE:
+        case ITEM00_ARROWS_SMALL:
+        case ITEM00_ARROWS_MEDIUM:
+        case ITEM00_ARROWS_LARGE:
+            break;
+        default:
+            return FALSE; // Proceed normally with the Get Item event.
+    }
+
+    ItemOverride override = ItemOverride_Lookup(&this->actor, gGlobalContext->sceneNum, GI_INVALID);
+    if (override.key.all == 0) {
+        return FALSE; // No override, proceed with Item_Give.
+    }
+
+    u16 resolvedItemId = ItemTable_ResolveUpgrades(override.value.itemId);
+    ItemRow* itemRow   = ItemTable_GetItemRow(resolvedItemId);
+
+    if ((itemRow->chestType != CHEST_JUNK && itemRow->chestType != CHEST_HEART) ||
+        itemRow->baseItemId == GI_HEART_PIECE || itemRow->baseItemId == GI_HEART_CONTAINER) {

```

Comment:
This doesn't include some items as major, like `GI_HEART_PIECE_WIN` or `GI_DOUBLE_DEFENSE`, and on the other hand it will always consider bombchus as such. Also, the `baseItemId` from the item row is not really the actual item, it mostly just affects the fanfare as explained in the comment above `rItemTable`.

```suggestion
    u8 isMajorItem = TRUE;
    switch (itemRow->chestType) {
        case CHEST_JUNK:
            isMajorItem = FALSE;
            break;
        case CHEST_BOMBCHUS:
            isMajorItem = gSaveContext.items[8] == 0xFF; // bombchus not obtained yet
            break;
        case CHEST_HEART:
            isMajorItem = resolvedItemId != GI_HEART;
            break;
    }

    if (isMajorItem) {
```